hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/kafka/schemas/SubjectsCompatibility.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2019, Logical Clocks AB. All rights reserved
+ *
+ * Hopsworks is free software: you can redistribute it and/or modify it under the terms of
+ * the GNU Affero General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * Hopsworks is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+package io.hops.hopsworks.common.dao.kafka.schemas;
+
+import io.hops.hopsworks.common.dao.project.Project;
+
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import javax.xml.bind.annotation.XmlRootElement;
+import java.io.Serializable;
+
+@Entity
+@Table(name = "subjects_compatibility",
+  catalog = "hopsworks",
+  schema = "")
+@XmlRootElement
+@NamedQueries({
+  @NamedQuery(name = "SubjectsCompatibility.setProjectCompatibility",
+    query = "UPDATE SubjectsCompatibility s SET s.compatibility = :compatibility " +
+      "WHERE s.subjectsCompatibilityPK.subject = 'projectcompatibility' AND s.project = :project"),
+  @NamedQuery(name = "SubjectsCompatibility.getProjectCompatibility",
+    query = "SELECT s FROM SubjectsCompatibility s WHERE s.project = :project AND " +
+      "s.subjectsCompatibilityPK.subject = 'projectcompatibility'"),
+  @NamedQuery(name = "SubjectsCompatibility.getSubjectCompatibility",
+    query = "SELECT s FROM SubjectsCompatibility s WHERE s.project = :project AND " +
+      "s.subjectsCompatibilityPK.subject = :subject"),
+  @NamedQuery(name = "SubjectsCompatibility.setSubjectCompatibility",
+    query = "UPDATE SubjectsCompatibility s SET s.compatibility = :compatibility " +
+      "WHERE s.subjectsCompatibilityPK.subject = :subject AND s.project = :project")})
+public class SubjectsCompatibility implements Serializable {
+  
+  @Basic(optional = false)
+  @NotNull
+  @Enumerated(EnumType.STRING)
+  @Column(name = "compatibility")
+  private SchemaCompatibility compatibility;
+  
+  @JoinColumn(name = "project_id",
+    referencedColumnName = "id",
+    insertable = false,
+    updatable = false)
+  @ManyToOne(optional = false)
+  private Project project;
+  
+  @EmbeddedId
+  private SubjectsCompatibilityPK subjectsCompatibilityPK;
```

Comment:
@kai-chi what's the motivation for not using an auto-increment PK?

Reply:
I could add an `id` auto-increment PK and set a unique key on the pair `(subject, project_id)`. The effect will be the same. Does it sound better to you?
table schema for ref: https://github.com/logicalclocks/hopsworks-chef/blob/b7275967346717ffc33110123cce29aae1309244/files/default/sql/ddl/1.1.0__initial_tables.sql#L1151